src/main/java/picard/analysis/CollectUmiPrevalenceMetrics.java

```diff
@@ -0,0 +1,181 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package picard.analysis;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.filter.*;
+import htsjdk.samtools.metrics.MetricsFile;
+import htsjdk.samtools.util.*;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.argparser.ExperimentalFeature;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import picard.cmdline.CommandLineProgram;
+import picard.cmdline.StandardOptionDefinitions;
+import picard.cmdline.programgroups.DiagnosticsAndQCProgramGroup;
+import picard.filter.CountingFilterWrapper;
+import picard.filter.CountingMapQFilter;
+import picard.filter.CountingPairedFilter;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.IntStream;
+
+import static picard.cmdline.StandardOptionDefinitions.MINIMUM_MAPPING_QUALITY_SHORT_NAME;
+
+/**
+ * A CLP that, tallies the number of UMIs in a duplicate set across the entire bam and produces a histogram.
+ *
+ * @author Yossi Farjoun
+ */
+@DocumentedFeature
+@ExperimentalFeature
+@CommandLineProgramProperties(
+        summary = CollectUmiPrevalenceMetrics.USAGE_SUMMARY + CollectUmiPrevalenceMetrics.USAGE_DETAILS,
+        oneLineSummary = CollectUmiPrevalenceMetrics.USAGE_SUMMARY,
+        programGroup = DiagnosticsAndQCProgramGroup.class
+)
+public class CollectUmiPrevalenceMetrics extends CommandLineProgram {
+
+    static final String USAGE_SUMMARY = "Tally the counts of UMIs in duplicate sets within a bam. \n";
+    static final String USAGE_DETAILS = "<p>" +
+            "This tool collects the Histogram of the number of duplicate sets that contain a given number of UMIs. " +
+            "Understanding this distribution can help understand the role that the UMIs have in the determination of " +
+            "consensus sets, the risk of UMI collisions, and of spurious reads that result from uncorrected UMIs.";
+
+
+    @Argument(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME, doc = "Input (indexed) BAM/CRAM file.")
+    public File INPUT;
```

Comment:
I hate to see new tools added that don't accept PicardHtsPath or at the very least Path.

Reply:
OUTPUT as File OK?

Reply:
Ideally it could more flexible, but if the metrics file infrastructure doesn't support that then I don't think we should try to overhaul it in this PR.